java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -157,79 +133,154 @@ void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
-      case Opcodes.CALOAD:
-      case Opcodes.DALOAD:
-      case Opcodes.FALOAD:
-      case Opcodes.IALOAD:
-      case Opcodes.LALOAD:
-      case Opcodes.SALOAD:
+      case IALOAD:
+      case LALOAD:
+      case FALOAD:
+      case DALOAD:
+      case AALOAD:
+      case BALOAD:
+      case CALOAD:
+      case SALOAD:
+        sv = constraintManager.createSymbolicValue(instruction);
+        programState = programState.unstackValue(2).state.stackValue(sv);
+        if (instruction.opcode != AALOAD) {
+          programState = programState.addConstraint(sv, ObjectConstraint.NOT_NULL);
+        }
         break;
-      case Opcodes.ASTORE:
-      case Opcodes.DSTORE:
-      case Opcodes.FSTORE:
-      case Opcodes.ISTORE:
-      case Opcodes.LSTORE:
+      case ISTORE:
+      case LSTORE:
+      case FSTORE:
+      case DSTORE:
+      case ASTORE:
         pop = programState.unstackValue(1);
         programState = pop.state.put(instruction.operand, pop.values.get(0));
         break;
-      case Opcodes.LDC:
-      case Opcodes.NEW: {
+      case IASTORE:
+      case LASTORE:
+      case FASTORE:
+      case DASTORE:
+      case AASTORE:
+      case BASTORE:
+      case CASTORE:
+      case SASTORE:
+        programState = programState.unstackValue(3).state;
+        break;
+      case POP:
+        programState = programState.unstackValue(1).state;
+        break;
+      case POP2:
+        programState = programState.unstackValue(2).state;
+        break;
+      case DUP:
+        sv = programState.peekValue();
+        Preconditions.checkNotNull(sv, "DUP on empty stack");
+        programState = programState.stackValue(sv);
+        break;
+      case DUP_X1:
+        pop = programState.unstackValue(2);
+        Preconditions.checkState(pop.values.size() == 2, "DUP_X1 needs 2 values on stack");
+        programState = pop.state.stackValue(pop.values.get(0)).stackValue(pop.values.get(1)).stackValue(pop.values.get(0));
+        break;
+      case DUP_X2:
+        pop = programState.unstackValue(3);
+        Preconditions.checkState(pop.values.size() == 3, "DUP_X2 needs 3 values on stack");
+        programState = pop.state.stackValue(pop.values.get(0)).stackValue(pop.values.get(2)).stackValue(pop.values.get(1)).stackValue(pop.values.get(0));
+        break;
+      case DUP2:
+        pop = programState.unstackValue(2);
+        Preconditions.checkState(pop.values.size() == 2, "DUP2 needs 2 values on stack");
+        programState = pop.state.stackValue(pop.values.get(1)).stackValue(pop.values.get(0)).stackValue(pop.values.get(1)).stackValue(pop.values.get(0));
+        break;
+      case DUP2_X1:
+        pop = programState.unstackValue(3);
+        Preconditions.checkState(pop.values.size() == 3, "DUP2_X1 needs 3 values on stack");
+        programState = pop.state.stackValue(pop.values.get(1)).stackValue(pop.values.get(0)).stackValue(pop.values.get(2)).stackValue(pop.values.get(1)).stackValue(pop.values.get(0));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 183 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)